KicadModTree/nodes/base/Pad.py

```diff
@@ -24,6 +24,105 @@
 from KicadModTree.nodes.base.Polygon import Polygon
 
 
+class RoundRadiusHandler(object):
+    r"""Handles round radius setting of a pad
+
+    :param \**kwargs:
+        See below
+
+    :Keyword Arguments:
+    * *radius_ratio* (``float``) --
+      The radius ratio of the rounded rectangle.
+    * *maximum_radius* (``float``) --
+      The maximum radius for the rounded rectangle.
+      If the radius produced by the radius_ratio parameter for the pad would
+      exceed the maximum radius, the ratio is reduced to limit the radius.
+      (This is useful for IPC-7351C compliance as it suggests 25% ratio with limit 0.25mm)
+    * *round_radius_exact* (``float``) --
+      Set an exact round radius for a pad.
+    """
+    def __init__(self, **kwargs):
+        self.radius_ratio = getOptionalNumberTypeParam(
+                            kwargs, 'radius_ratio', default_value=0,
+                            low_limit=0, high_limit=0.5)
+
+        self.maximum_radius = getOptionalNumberTypeParam(kwargs, 'maximum_radius')
+        self.round_radius_exact = getOptionalNumberTypeParam(kwargs, 'round_radius_exact')
+
+        self.kicad4_compatible = kwargs.get('kicad4_compatible', False)
+
+    def getRadiusRatio(self, shortest_sidelength):
+        r"""get the resulting round radius ratio
+
+        :param shortest_sidelength: shortest sidelength of a pad
+        :return: the resulting round radius ratio to be used for the pad
+        """
+        if self.kicad4_compatible:
+            return 0
+
+        if self.round_radius_exact:
+            if self.round_radius_exact > shortest_sidelength/2:
+                raise ValueError(
+                    "requested round radius of {} is too large for pad size of {}"
+                    .format(self.round_radius_exact, pad_size)
+                    )
+            if self.maximum_radius:
+                return min(self.round_radius_exact, self.maximum_radius)/shortest_sidelength
+            else:
+                return self.round_radius_exact/shortest_sidelength
+        if self.maximum_radius:
+            if self.radius_ratio*shortest_sidelength > self.maximum_radius:
+                return self.maximum_radius/shortest_sidelength
+
+        return self.radius_ratio
+
+    def getRoundRadius(self, shortest_sidelength):
+        r"""get the resulting round radius
+
+        :param shortest_sidelength: shortest sidelength of a pad
+        :return: the resulting round radius to be used for the pad
+        """
+        return self.getRadiusRatio(shortest_sidelength)*shortest_sidelength
+
+    def roundingRequested(self):
+        r"""Check if the pad has a rounded corner
+
+        :return: True if rounding corners is required
```

Comment:
"is" should be "are".

Reply:
fixed

Reply:
Change "is" to "are", not "if" to "are". The wrong word was changed.